components/org.wso2.carbon.identity.oauth.endpoint/src/main/java/org/wso2/carbon/identity/oauth/endpoint/util/EndpointUtil.java

```diff
@@ -1804,4 +1809,72 @@ public static boolean isExternalConsentPageEnabledForSP(ServiceProvider serviceP
     public static boolean isConsentPageRedirectParamsAllowed() {
         return FileBasedConfigurationBuilder.getInstance().isConsentPageRedirectParamsAllowed();
     }
+
+    /**
+     * Returns an instance of OAuthAuthzRequest. If the configured classname is invalid the default implementation
+     * will be returned.
+     *
+     * @param request http servlet request.
+     * @return instance of OAuthAuthzRequest.
+     * @throws OAuthProblemException thrown when initializing the OAuthAuthzRequestClass instance.
+     * @throws OAuthSystemException  thrown when initializing the OAuthAuthzRequestClass instance.
+     */
+    public static OAuthAuthzRequest getOAuthAuthzRequest(HttpServletRequest request)
+            throws OAuthProblemException, OAuthSystemException {
+
+        if (isDefaultOAuthAuthzRequestClassConfigured()) {
+            return new CarbonOAuthAuthzRequest(request);
+        }
+        try {
+            Class<? extends OAuthAuthzRequest> clazz = getOAuthAuthzRequestClass();
+            // Validations will be performed when initializing the class instance.
+            Constructor<?> constructor = clazz.getConstructor(HttpServletRequest.class);
+            return (OAuthAuthzRequest) constructor.newInstance(request);
+        } catch (InvocationTargetException e) {
+            // Handle OAuthProblemException & OAuthSystemException thrown from extended class.
+            if (e.getTargetException() instanceof OAuthProblemException) {
+                throw (OAuthProblemException) e.getTargetException();
+            } else if (e.getTargetException() instanceof OAuthSystemException) {
+                throw (OAuthSystemException) e.getTargetException();
+            } else {
```

Comment:
we can move the log outside the else block and then remove the empty else block right?

Reply:
then the same log will be logged twice if the other types of exceptions are caught